packages/peregrine/lib/talons/MyAccount/useNewsletterSubscription.js

```diff
@@ -0,0 +1,73 @@
+import { useCallback, useState } from 'react';
+import { useMutation } from '@apollo/react-hooks';
+import { useToasts } from '@magento/peregrine';
+import { useUserContext } from '../../context/user';
+import { useAwaitQuery } from '../../hooks/useAwaitQuery';
+
+export const useNewsletterSubscription = props => {
+    const { subscribeMutation, customerQuery } = props;
+    const [{ currentUser, isSignedIn }, { getUserDetails }] = useUserContext();
```

Comment:
Pull out user slice data and action in favor of running your own Apollo query. You may continue using `isSignedIn` for now, that will likely be moved to the `UserContext`.